flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -2138,12 +2080,10 @@ private void reportToStatsTracker(
         }
         Map<JobVertexID, Integer> vertices =
                 tasks.values().stream()
-                        .map(ExecutionVertex::getJobVertex)
-                        .distinct()
                         .collect(
-                                toMap(
-                                        ExecutionJobVertex::getJobVertexId,
-                                        ExecutionJobVertex::getParallelism));
+                                Collectors.groupingBy(
+                                        ExecutionVertex::getJobvertexId,
+                                        Collectors.reducing(0, e -> 1, Integer::sum)));
```

Comment:
I'm interested to know why this was necessary.

Reply:
This change is doing a preparation for the case that are some tasks are finished. In this case the `tasksToWaitFor` might not contains all the job vertices and we need to compute based on both `tasksToWaitFor` and `finishedTasks`. But it should be more suitable to be changed together in the next PR, so I'll revert it first.

Reply:
👍